release.sh

```diff
@@ -57,6 +57,14 @@ RELEASE_VERSION_BUILDMETADATA="$(echo "$RELEASE_VERSION" | perl -pe "s/${semver_
 # Setup
 #
 
+#serialize an object containing repo credentials to disk, and put the path to it in an environment variable
+if [ ! -z "$ARTIFACTORY_USER" ] && [ ! -z "$ARTIFACTORY_TOKEN" ]; then
+    export REPOCREDSPATH="/tmp/"
+    export REPOCREDSFILENAME="repo_creds.json"
+    export REPOCREDSFULL=$REPOCREDSPATH$REPOCREDSFILENAME
+    jq --null-input   --arg url "sles-mirror" --arg realm "Artifactory Realm" --arg user "$ARTIFACTORY_USER"   --arg password "$ARTIFACTORY_TOKEN"   '{($url): {"realm": $realm, "user": $user, "password": $password}}' > $REPOCREDSFULL
+fi
+
```

Comment:
So it looks like `/tmp/repo_creds.json` will be left on jenkins runner? Is there any way to clean it up inevitably (e.g. even if Jnekins job gets cancelled)? Ideally we should not write sensitive information to filesystem, just pass username and password into docker container via environment variables.

Reply:
I added code to Jenkinsfile.github to remove /tmp/repo_creds.json in the post release section, either success or fail

Reply:
I'm afraid this does not cover all cases (for example, aborted builds). Also hard to maintain, not clear where this file is getting created. IMO better just add 

    trap "rm -f '${REPOCREDSFULL}'" EXIT

after line 65 in `release.sh`.